rustls/src/msgs/base.rs

```diff
@@ -57,6 +57,36 @@ impl<'a> DerefMut for BorrowedPayload<'a> {
 }
 
 impl<'a> BorrowedPayload<'a> {
+    pub fn truncate(&mut self, len: usize) {
+        if len >= self.len() {
+            return;
+        }
+
+        self.0 = core::mem::take(&mut self.0)
+            .split_at_mut(len)
+            .0;
+    }
+
+    #[cfg(test)]
+    pub(crate) fn new(bytes: &'a mut [u8]) -> Self {
+        Self(bytes)
+    }
+
+    pub(crate) fn into_inner(self) -> &'a mut [u8] {
+        self.0
+    }
+
+    pub(crate) fn pop(&mut self) -> Option<u8> {
+        if self.is_empty() {
+            None
+        } else {
+            let len = self.len();
+            let last = self[len - 1];
+            self.truncate(len - 1);
+            Some(last)
+        }
+    }
```

Comment:
Style nit: this is a pretty surprising method order IMO.

Reply:
Reordered